packages/sdk/src/interaction/interaction.ts

```diff
@@ -272,29 +262,31 @@ const recPipe = async (ix, fns = []) => {
   }
 }
 
-export const pipe = (...args) => {
+export const pipe = (...args: any[]) => {
   const [arg1, arg2] = args
-  if (isArray(arg1) && arg2 == null) return d => pipe(d, arg1)
+  if (isArray(arg1) && arg2 == null) return (d: any) => pipe(d, arg1)
   return recPipe(arg1, arg2)
 }
 
-const identity = v => v
+const identity = (v: any, ..._: any[]) => v
```

Comment:
```suggestion
const identity = <T>(v: T) => v
```
Can we strongly type this?  And is the rest parameter necessary?